### PR TITLE
Implement noopWindowsResourceAllocator

### LINKED ISSUE #88913: Pods not starting on Windows nodes
**What happened**:

Pods will not start on Windows nodes, even though nodes are ready and pods are scheduled to nodes.

Kubelet logs are filled with this:

```
E0306 18:20:36.813221    3564 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 206 [running]:
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x38ac960, 0x6b90c30)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xaa
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x89
panic(0x38ac960, 0x6b90c30)
	/usr/local/go/src/runtime/panic.go:679 +0x1c0
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).canAdmitPod(0xc000640a80, 0x0, 0x0, 0x0, 0xc00102d400, 0x0, 0x0, 0x6a62da3, 0xa, 0x774)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1794 +0x15d
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).HandlePodAdditions(0xc000640a80, 0xc000d5e1a8, 0x1, 0x1)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:2071 +0x312
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).syncLoopIteration(0xc000640a80, 0xc000680ae0, 0x48cc5c0, 0xc000640a80, 0xc000816ea0, 0xc000816f00, 0xc0009f1800, 0xc0002f0601)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1913 +0x561
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).syncLoop(0xc000640a80, 0xc000680ae0, 0x48cc5c0, 0xc000640a80)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1856 +0x378
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).Run(0xc000640a80, 0xc000680ae0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1452 +0x26e
k8s.io/kubernetes/cmd/kubelet/app.startKubelet.func1()
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubelet/app/server.go:1133 +0x43
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000b352a0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x65
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000b352a0, 0x4824e60, 0xc000b39cb0, 0xc0000bf601, 0xc0000800c0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xaa
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000b352a0, 0x0, 0x0, 0x1, 0xc0000800c0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0xe9
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc000b352a0, 0x0, 0xc0000800c0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x54
created by k8s.io/kubernetes/cmd/kubelet/app.startKubelet
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

Build K8s from master, create any pod on Windows node.

**Anything else we need to know?**:

I strongly believe this commit is responsible: https://github.com/kubernetes/kubernetes/commit/ac32644d6e712af0747f39a945c8df29d6a9cfc4 

What happened is that now, with the commit mentioned above, a nil podAdmitHandler is registered, that when called will result in a panic.

More specifically here: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L871

For Windows, `GetAllocateResourcesPodAdmitHandler` returns `nil`:

https://github.com/kubernetes/kubernetes/blob/1afb554222605d4d67a911c9bbb0fa5b1049c2dc/pkg/kubelet/cm/container_manager_windows.go#L181

**Environment**:

All K8s built from master. This affects all sig-windows master tests.

### PR DESCRIPTION
**Pull request by adelina-t**
On Windows, the podAdmitHandler returned by the GetAllocateResourcesPodAdmitHandler() func
and registered by the Kubelet is nil.

We should check for nil before registering any podAdmitHandlers.

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change

/kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**Which issue(s) this PR fixes**:

Fixes #88913

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by benmoss** [comment]
@liggitt this is more fallout from #87759

**Comment by liggitt** [comment]
Is there a test that demonstrates the bug this fixes?

**Comment by adelina-t** [comment]
@liggitt ^

**Review by michmike — APPROVED** [review]
/lgtm
/approve

**Comment by dashpole** [comment]
/lgtm
/priority critical-urgent
looks urgent to me, since it causes failure to start on windows.

**Comment by liggitt** [comment]
/assign @derekwaynecarr 
as approver of https://github.com/kubernetes/kubernetes/pull/87759

**Comment by pjh** [comment]
/lgtm

Thanks Adelina!

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

Long term, another option would be for `AddPodAdmitHandler` to gracefully handle being passed `nil`? Definitely not blocking but curious if you thought about going that route?

**Comment by liggitt** [comment]
given the test windows cluster has failed to start on multiple consecutive runs, is there further investigation that needs to be done before retest?

**Comment by adelina-t** [comment]
Windows cluster failed over the weekend two times because of staging error (?) , nothing to do with the cluster:

```
W0308 21:07:52.765] 2020/03/08 21:07:52 process.go:153: Running: /go/src/k8s.io/release/push-build.sh --nomock --verbose --noupdatelatest --bucket=kubernetes-release-pull --ci --gcs-suffix=/pull-kubernetes-e2e-aks-engine-azure-windows --allow-dup
I0308 21:07:52.965] push-build.sh: BEGIN main on 6af94723-617f-11ea-b188-56b122f30985 Sun Mar  8 21:07:52 UTC 2020
I0308 21:07:52.965] 
W0308 21:07:53.066] tar (child): /go/src/k8s.io/kubernetes/_output/release-tars/kubernetes.tar.gz: Cannot open: No such file or directory
W0308 21:07:53.066] tar (child): Error is not recoverable: exiting now
W0308 21:07:53.067] tar: Child returned status 2
W0308 21:07:53.067] tar: Error is not recoverable: exiting now
I0308 21:07:53.167] 
I0308 21:07:53.168] 
I0308 21:07:53.168] Signal ERR caught!
```

Today it failed because a timeout in installing features on one of the Windows nodes:

```
Add-WindowsFeature : The server could not update the provided feature files in the time allowed.
```

None of these failures are related to this code.

**Comment by adelina-t** [comment]
With latest Windows presubmit run, Kubelet starts on Windows nodes, so this PR is doing it's job. The test failures are related to the fact that there is a missing agnhost image: e2eteam/agnhost:2.12 . @claudiubelu  just pushed it so I'll triggered a retest.

/retest

**Comment by chewong** [comment]
@adelina-t I think https://github.com/kubernetes/test-infra/pull/16537 broke the presubmit job. Let me modify the job configuration accordingly.

**Comment by adelina-t** [comment]
There are only 2 tests failing in the latest presubmit aks-engine job run, both of them unrelated to this patch. We should really merge this so we can have the Kubelet up and running ( and the e2e tests ) so we can determine if the two failing tests are flakes or if they are really an issue.

**Comment by liggitt** [comment]
GetAllocateResourcesPodAdmitHandler() is not documented to be able to return nil, and all other implementations return a non-nil interface. It seems less fragile to have the windows version return a no-op implementation as well

**Comment by michmike** [comment]
@liggitt, @adelina-t  will make the change you requested. Once done, we would appreciate a quick approval since this fix impacts a lot of windows tests that are failing.

File: pkg/kubelet/cm/container_manager_windows.go
Comment (liggitt) [inline pkg/kubelet/cm/container_manager_windows.go]: `hack/update-gofmt.sh` might not be happy with the whitespace change

**Comment by liggitt** [comment]
/approve
/lgtm

change looks good to me. check if `hack/update-gofmt.sh` fixes the whitespace and push an update to fix that if so

**Comment by liggitt** [comment]
also, per https://github.com/kubernetes/kubernetes/pull/88917#issuecomment-597243384, drop `Fixes: #88913` from the commit description (you can move it into the PR description)

**Comment by neolit123** [comment]
/lgtm
based on the latest force push diff:
https://github.com/kubernetes/kubernetes/compare/e2caa2b93cab0ce21df6e18f5775323641d8ac35..a9f834d17dc990bd3ba282b51d784a567ecb86d6

/restest

**Pull request merged**
